src/common/hooks/use-do-change-screen.ts

```diff
@@ -0,0 +1,29 @@
+import { useDispatch } from 'react-redux';
+import { doChangeScreen as changeScreen } from '@store/onboarding/actions';
+import { ScreenPaths } from '@store/onboarding/types';
+import { useNavigate } from 'react-router-dom';
+import { useCallback } from 'react';
+
+type doChangeScreen = (path: ScreenPaths, changeRoute?: boolean) => void;
+
+export function useDoChangeScreen(): doChangeScreen {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const doNavigatePage = useCallback(
+    (path: ScreenPaths) => {
+      navigate(path);
+      dispatch(changeScreen(path));
+    },
+    [dispatch, navigate]
+  );
+
+  return useCallback(
+    (path: ScreenPaths, changeRoute = true) => {
+      if (changeRoute) {
+        return doNavigatePage(path);
+      }
+    },
+    [doNavigatePage]
+  );
```

Comment:
Better to return two functions, than provide a flag to modify one function's behaviour.

Reply:
I agree -- however the goal of this PR is to just remove the analytics -- I want to refactor how routing is done more generally, and this kind of change would fit well in that kind of PR.